packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -382,6 +382,30 @@ abstract class RunCommandBase extends FlutterCommand with DeviceBasedDevelopment
       );
     }
   }
+
+  Future<WebDevServerConfig> webDevServerConfigCore() async {
```

Comment:
@Saqib198 – I moved things into this shared function so we have fewer places to keep this logic up-to-date!